### PR TITLE
TF upgrade to 5.15.1

### PR DESCRIPTION
**Pull request by quic-mamta**
## Summary

  This PR upgrades QEfficient’s Transformers dependency from `5.5.4` to `5.15.1` and updates the affected model wrappers/tests for API changes introduced in the newer Transformers release.

  ## Changes

  - Updated the Transformers pin to `5.15.1`.
  - Added compatibility for renamed/moved Transformers classes:
    - `CLIPTextTransformer` fallback to `CLIPTextModel`
    - GraniteMoE expert/router rename handling
    - Gemma3 logger import change
  - Updated QEff cache compatibility by adding `get_max_length()` on `QEffDynamicLayer`.
  - Fixed Gemma4 heterogeneous config handling:
    - Avoid direct global access to per-layer attributes like `head_dim`/`global_head_dim`
    - Resolve per-layer config values when building dummy KV cache tensors
    - Updated Gemma4 accuracy tests to avoid unsafe `hasattr(...)` on heterogeneous configs
  - Updated Qwen3.5 and Qwen3.5-MoE wrappers for Transformers 5.15.1 recurrent attention mask/layer-type behavior.
  - Fixed GLM4-MoE router reshaping to use `hidden_dim`.
  - Updated transform coverage for renamed optimized MoE components.

### CODE REVIEW

File: QEfficient/transformers/models/clip/modeling_clip.py [hunk 1]
```diff
@@ -18,11 +18,13 @@
...
-from typing import Optional
-
 import torch
 from transformers.modeling_outputs import BaseModelOutputWithPooling
-from transformers.models.clip.modeling_clip import CLIPTextTransformer
+
+try:
+    from transformers.models.clip.modeling_clip import CLIPTextTransformer
+except ImportError:
+    from transformers.models.clip.modeling_clip import CLIPTextModel as CLIPTextTransformer
```
Comment (ochougul) [inline QEfficient/transformers/models/clip/modeling_clip.py]: ?

**Comment by quic-mamta** [comment]
CI-Ready
